### ISSUE TITLE
Insta-crash when opening large folder

### ISSUE THREAD

**Issue by Mrnofish** [flackbash/AudioAnchor#181]
I have a podcasts of sorts that uses no standard podcasting technology, it was scraped manually and is a single folder containing over 300 hundred files.

Previously, this folder used to work in AudioAnchor albeit slowly. Nowadays, the app will crash as soon as the folder is entered from the library screen.

May be related to #167. I'm also on Android 11 (LOS 18.1). It's been a while since I last tried opening this particular folder in AudioAnchor and I'm not entirely sure whether, back then, the OS had already been upgraded.

**Comment by J0J0** [comment]
Based on the available information, i'm unable to reproduce this bug with
* my phone (Android 10)
* emulator Android 11
* v2.3.2 (last realease on f-droid?!)

nor with
* master branch at 51660c71a25b1a35395e8dbebfe7ca49526a4a08 against emulator Android 11.

I can open albums with 300, 500, 1000 and 5000 files almost without delay. I used mp3 files of 1 second length.
I also have audiobooks with more than 500 files, which never caused me problems so far (all mp3, though).

**Comment by Mrnofish** [comment]
I can still reproduce just fine, and rolling back to 2.3.0 as reported in the OP and in #167 does indeed fix the issue.

> v2.3.2 (last realease on f-droid?!)
Well, since it is the latest release...

Flackbash already wrote she was having trouble reproducing in the emu, and that the issue is likely related to changes in Android 11, so I'm afraid testing under circumstances that are markedly different from the ones reported is kind of a waste of time.

Thanks for trying, though.

**Comment by lexillar** [comment]
I have a smaller, but still large library of .opus files on weak phone with 32gb memory.

The problem is that when it scans (or accidentally scans after missswap upwards) the program and ui becomes irresponsive for the time longer than crash manager proposes to kill or wait, longer than phone's backlight off. 

It would be better to scan in parallel low priority process, so ui remains responsive.

**Comment by reaseno** [comment]
I have the same problem, but the app doesn't crash for me; it just freezes for a long time until it has re-scanned the directories. I have directories containing MP3 files, but also directories that only contain other directories with MP3 files. One such top-level directory represents a book series, with the respective books contained in the subdirectories.
